iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/AbstractNewResourceCheck.java

```diff
@@ -104,4 +103,26 @@ private Pattern pattern(String regex, int flags) {
   public Predicate<ExpressionTree> isTrue() {
     return TextUtils::isValueTrue;
   }
+
+  public static boolean isResource(BlockTree blockTree) {
+    return "resource".equals(blockTree.key().value());
+  }
+
+  public static boolean isResourceOfType(BlockTree blockTree, String resourceType) {
+    return "resource".equals(blockTree.key().value()) && resourceType.equals(resourceType(blockTree));
+  }
+
+  public static boolean isData(BlockTree blockTree) {
+    return "data".equals(blockTree.key().value());
+  }
+
```

Comment:
Both methods are only used in test cases and even not fully covered. Do we need them right now? I think we can add them later, when we need them.